components/generic/w-toggle-item.vue

```diff
@@ -1,23 +1,64 @@
 <template>
-  <input :id="id" v-model="model" v-bind="$attrs" />
-  <label v-if="label" :for="id" v-html="label" :class="labelClass" />
-  <label v-else :for="id" :class="labelClass"><slot /></label>
+  <input 
+    :id="id" 
+    v-model="model" 
+    :type="type"
+    :radioButton="radioButton"
+    :disabled="disabled"
+    :invalid="invalid"
+    :equalWidth="equalWidth"
+    v-bind="$attrs" 
+    :class="[inputClasses]" 
+  />
+  <label v-if="label" :for="id" v-html="label" :class="labelClasses" />
+  <label v-else :for="id" :class="labelClasses"><slot /></label>
 </template>
 
-<script>
-import { id } from '#util'
-import { modelProps, createModel } from 'create-v-model'
+<script setup>
+import { computed } from 'vue';
+import { id as uniqueId } from '#util';
+import { modelProps, createModel } from 'create-v-model';
+import { toggle as ccToggle } from '@warp-ds/component-classes';
+const p = defineProps({
+  id: { ...uniqueId },
+  label: String,
+  type: String,
+  class: String,
+  disabled: Boolean,
+  equalWidth: Boolean,
+  invalid: Boolean,
+  radioButton: Boolean,
+  labelClass: null,
+  ...modelProps()
+});
+
+const emit = defineEmits(['update:modelValue']);
+const model = createModel({ props: p, emit });
+
+const isRadio = computed(() => p.type === 'radio');
+const isCheckbox = computed(() => p.type === 'checkbox');
+const labelClasses = computed(() => (p.labelClass || {
+  [ccToggle.label]: !p.radioButton,
+  [`${ccToggle.labelDisabled} ${isCheckbox.value ? ccToggle.checkboxDisabled : ccToggle.radioDisabled}`] : p.disabled,
+  [ccToggle.focusable]: !p.radioButton,
+  [ccToggle.noContent]: !p.radioButton,
+  [ccToggle.radioDisabled]: isRadio.value && p.disabled,
+  [ccToggle.labelRadioColors] : isRadio.value && !p.radioButton,
+  [ccToggle.radio]: isRadio.value,
+  [ccToggle.radioInvalid]: isRadio.value && p.invalid,
+  [ccToggle.checkboxInvalid]: isCheckbox.value && p.invalid,
+  [`${ccToggle.checkbox} ${ccToggle.labelCheckboxColors} ${ccToggle.icon} ${p.disabled ? '' : ccToggle.checkboxChecked}`]: isCheckbox.value,
+  [ccToggle.scLabel]: p.radioButton,
+  [ccToggle.scLabelJustified]: p.equalWidth,
+  [ccToggle.scLabelSmall]: p.small,
+}));
+const inputClasses = {
+  [p.class || ccToggle.input]: true,
```

Comment:
Why is it either props class or `ccToggle.input` ("sr-only peer")? The "sr-only" bit I understand but is it safe to remove `peer` class if another class comes in props?

Reply:
I would at least not mind having both set if `p.class` exists 😄

Reply:
Agree, let's have it like that